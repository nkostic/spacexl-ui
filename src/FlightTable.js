/* eslint react/no-did-mount-set-state: 0 */
import React, { PureComponent } from 'react';
import FlightRow from './FlightRow';

class FlightTable extends PureComponent {
  state = {
    flights: [],
  }

  async componentDidMount() {
    try {
      const res = await fetch('http://localhost:3000/api/get/flights');
      const flights = await res.json();
      this.setState({
        flights: flights.results,
      });
    } catch (e) {
      console.log(e);
    }
  }

  render() {
    return (
      <div>
        {this.state.flights.map(flight => <FlightRow key={flight.id} flight={flight} />)}
      </div>
    );
  }
}

export default FlightTable;