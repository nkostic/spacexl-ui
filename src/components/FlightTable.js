/* eslint react/no-did-mount-set-state: 0 */
import React, { PureComponent } from "react";
import axios from 'axios';
import FlightTableRow from "./FlightTableRow";
import Spinner from 'react-spinkit';

class FlightTable extends PureComponent {

  constructor(props) {
    super(props);

    this.state = {
      flights: [],
      land: false,
      reused: false,
      with: false,
      loading : true
    }

    this.fetch = this.fetch.bind(this);
    this.handleChange = this.handleChange.bind(this);
  }

  fetch() {
    try {

      this.setState({
        loading: true
      });

      const filters = {
        land: this.state.land,
        reused: this.state.reused,
        with: this.state.with
      }

      axios.post(`${process.env.REACT_APP_API_URL}/api/get/flights`, { filters })
      .then(res => {
        const flights = res.data.results;
        this.setState({
          flights,
          loading: false
        });
      })

    } catch (e) {
      console.log(e);
    }
  }

  async componentDidMount() {
    this.fetch();
  }

  handleChange(e) {
    const item = e.target.name;
    const isChecked = e.target.checked;
    switch (item) {
      case 'land':
        this.setState({'land':isChecked});
        break;
      case 'reused':
        this.setState({'reused':isChecked});
        break;
      case 'with':
        this.setState({'with':isChecked});
        break;
      default:
        break;
    }
  }

  render() {
    return (
      <section className="section">
        <div className="container">
          <div className="columns filters">
            <div className="column">
              <div className="level">
                <div className="level-left is-inline-mobile">
                  <div className="level-item is-inline-mobile">
                    <button
                      onClick={this.fetch}
                      className="button refresh"
                    ></button>
                  </div>
                </div>
                <div className="level-right is-inline-mobile">
                  <div className="level-item is-inline-mobile">
                    <label className="checkbox">
                      LAND
                      <input type="checkbox" 
                        name="land"
                        onChange={this.handleChange}
                        checked={this.state.land}
                      />
                      <span className="checkmark"></span>
                    </label>
                  </div>
                  <div className="level-item is-inline-mobile">
                    <label className="checkbox">
                      REUSED
                      <input type="checkbox" 
                        name="reused"
                        onChange={this.handleChange}
                        checked={this.state.reused}
                      />
                      <span className="checkmark"></span>
                    </label>
                  </div>
                  <div className="level-item is-inline-mobile">
                    <label className="checkbox">
                      <span className="is-hidden-mobile">WITH </span>
                      <span>REDDIT</span>
                      <input type="checkbox" 
                        name="with"
                        onChange={this.handleChange}
                        checked={this.state.with}
                      />
                      <span className="checkmark"></span>
                    </label>
                  </div>
                </div>
              </div>
            </div>
          </div>

          <div className="columns table-header is-hidden-mobile">
            <div className="column">
              <div className="columns">
                <div className="column is-1">Badge</div>
                <div className="column">Rocket Name</div>
                <div className="column">Rocket Type</div>
                <div className="column">Launch Date</div>
                <div className="column is-4">Details</div>
                <div className="column is-1">ID</div>
                <div className="column is-1">Article</div>
              </div>
            </div>
          </div>
          
          {this.state.flights.map(flight => (
            <FlightTableRow key={flight.id} flight={flight} />
          ))}

          {this.state.loading && 
            <div className="columns is-centered">
              <div className="column has-text-centered is-1 loading">
                <Spinner name='ball-scale-multiple' color="white"/>
              </div>
            </div>
          }
          
        </div>
      </section>
    );
  }
}

export default FlightTable;
