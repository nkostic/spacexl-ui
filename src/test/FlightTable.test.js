import React from 'react';
import { render, cleanup } from '@testing-library/react';
import FlightTable from '../components/FlightTable';

afterEach(cleanup)

test('<FlightTable/>', () => {
  const { getByText, container } = render(<FlightTable />);
  const labeLand = getByText(/LAND/i);
  const labeReused = getByText(/REUSED/i);
  const labelWith = getByText(/WITH/i);
  const headerBadge = getByText(/Badge/i);
  const headerRocketName = getByText(/Rocket Name/i);
  const headerRocketType = getByText(/Rocket Type/i);
  const headerLaunchDate = getByText(/Launch Date/i);
  const headerDetails= getByText(/Details/i);
  const headerID= getByText(/ID/i);
  const headerArticle= getByText(/Article/i);
  expect(labeLand).toBeInTheDocument();
  expect(labeReused).toBeInTheDocument();
  expect(labelWith).toBeInTheDocument();
  expect(headerBadge).toBeInTheDocument();
  expect(headerRocketName).toBeInTheDocument(); 
  expect(headerRocketType).toBeInTheDocument(); 
  expect(headerLaunchDate).toBeInTheDocument(); 
  expect(headerDetails).toBeInTheDocument(); 
  expect(headerID).toBeInTheDocument(); 
  expect(headerArticle).toBeInTheDocument(); 
  expect(container.firstChild).toMatchSnapshot();
});
