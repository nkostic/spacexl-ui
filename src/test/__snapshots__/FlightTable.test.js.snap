// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FlightTable/> 1`] = `
<section
  class="section"
>
  <div
    class="container"
  >
    <div
      class="columns filters"
    >
      <div
        class="column"
      >
        <div
          class="level"
        >
          <div
            class="level-left is-inline-mobile"
          >
            <div
              class="level-item is-inline-mobile"
            >
              <button
                class="button refresh"
              />
            </div>
          </div>
          <div
            class="level-right is-inline-mobile"
          >
            <div
              class="level-item is-inline-mobile"
            >
              <label
                class="checkbox"
              >
                LAND
                <input
                  name="land"
                  type="checkbox"
                />
                <span
                  class="checkmark"
                />
              </label>
            </div>
            <div
              class="level-item is-inline-mobile"
            >
              <label
                class="checkbox"
              >
                REUSED
                <input
                  name="reused"
                  type="checkbox"
                />
                <span
                  class="checkmark"
                />
              </label>
            </div>
            <div
              class="level-item is-inline-mobile"
            >
              <label
                class="checkbox"
              >
                <span
                  class="is-hidden-mobile"
                >
                  WITH 
                </span>
                <span>
                  REDDIT
                </span>
                <input
                  name="with"
                  type="checkbox"
                />
                <span
                  class="checkmark"
                />
              </label>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="columns table-header is-hidden-mobile"
    >
      <div
        class="column"
      >
        <div
          class="columns"
        >
          <div
            class="column is-1"
          >
            Badge
          </div>
          <div
            class="column"
          >
            Rocket Name
          </div>
          <div
            class="column"
          >
            Rocket Type
          </div>
          <div
            class="column"
          >
            Launch Date
          </div>
          <div
            class="column is-4"
          >
            Details
          </div>
          <div
            class="column is-1"
          >
            ID
          </div>
          <div
            class="column is-1"
          >
            Article
          </div>
        </div>
      </div>
    </div>
    <div
      class="columns is-centered"
    >
      <div
        class="column has-text-centered is-1 loading"
      >
        <div
          class="sk-fade-in sk-spinner ball-scale-multiple"
          color="white"
          style="color: white;"
        >
          <div />
          <div />
          <div />
        </div>
      </div>
    </div>
  </div>
</section>
`;
